src/Okta.Sdk.IntegrationTests/TrustedOriginScenarios.cs

```diff
@@ -0,0 +1,241 @@
+﻿// <copyright file="TrustedOriginScenarios.cs" company="Okta, Inc">
+// Copyright (c) 2020 - present Okta, Inc. All rights reserved.
+// Licensed under the Apache 2.0 license. See the LICENSE file in the project root for full license information.
+// </copyright>
+
+using FluentAssertions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Okta.Sdk.IntegrationTests
+{
+    public class TrustedOriginScenarios
+    {
+        private static string dotnetSdkPrefix = "dotnetSdk";
+
+        [Fact]
+        public async Task ListOrigins()
+        {
+            var client = TestClient.Create();
+            var trustedOrigins = await client.TrustedOrigins.ListOrigins().ToListAsync();
+            trustedOrigins.Should().NotBeNull();
+            trustedOrigins.Count.Should().BeGreaterThan(0);
+        }
+
+        [Fact]
+        public async Task CreateOrigin()
+        {
+            var client = TestClient.Create();
+            var testTrustedOriginName = $"{dotnetSdkPrefix}_{nameof(CreateOrigin)}_Test";
```

Comment:
Do we need a random keyword to avoid name collisions?

Reply:
Good idea; my thought was that would only be necessary if the finally block below was somehow not reached.  However, multiple instances of this test may run simultaneously so I will put in the random keyword as you suggest.